src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py

```diff
@@ -93,6 +94,25 @@ def _complex_params(command_group):
             c.argument('start_ip_address', options_list=['--start-ip-address'], help='The start IP address of the firewall rule. Must be IPv4 format. Use value \'0.0.0.0\' to represent all Azure-internal IP addresses.')
             c.argument('end_ip_address', options_list=['--end-ip-address'], help='The end IP address of the firewall rule. Must be IPv4 format. Use value \'0.0.0.0\' to represent all Azure-internal IP addresses.')
 
+    for scope in ['mysql server vnet-rule', 'postgres server vnet-rule']:
+        with self.argument_context(scope) as c:
+            c.argument('server_name', options_list=['--server-name', '-s'])
+            c.argument('virtual_network_rule_name', options_list=['--name', '-n'], id_part='child_name_1', help='The name of the vnet rule.')
+            c.argument('virtual_network_subnet_id', options_list=['--subnet'], help='Name or ID of the subnet that allows access to an Azure Postgres Server. If subnet name is provided, --vnet-name must be provided.')
+            c.argument('ignore_missing_vnet_service_endpoint', options_list=['--ignore-missing-endpoint', '-i'], help='Create vnet rule before virtual network has vnet service endpoint enabled', arg_type=get_three_state_flag())
+
+        with self.argument_context('postgres server vnet-rule create') as c:
+            c.extra('vnet_name', options_list=['--vnet-name'], help='The virtual network name', validator=validate_subnet)
+
+        with self.argument_context('postgres server vnet-rule update') as c:
+            c.extra('vnet_name', options_list=['--vnet-name'], help='The virtual network name', validator=validate_subnet)
+
+        with self.argument_context('mysql server vnet-rule create') as c:
+            c.extra('vnet_name', options_list=['--vnet-name'], help='The virtual network name', validator=validate_subnet)
+
+        with self.argument_context('mysql server vnet-rule update') as c:
+            c.extra('vnet_name', options_list=['--vnet-name'], help='The virtual network name', validator=validate_subnet)
```

Comment:
To avoid the duplication, you could use:

```Python
for scope in ['postgres server vnet-rule create', 'postrgres server vnet-rule update', 'mysql server vnet-rule create', 'mysql server vnet-rule update']:

  with self.argument_context(scope) as c:
   c.extra(…)
```

Reply:
done.